data/transactions/logic/eval.go

```diff
@@ -1152,12 +1146,10 @@ func EvalContract(program []byte, gi int, aid basics.AppIndex, params *EvalParam
 			}
 		}
 		// and add the appID to `createdApps`
-		if cx.EvalParams.Proto.LogicSigVersion >= sharedResourcesVersion {
-			if cx.EvalParams.available.createdApps == nil {
-				cx.EvalParams.available.createdApps = make(map[basics.AppIndex]struct{})
-			}
-			cx.EvalParams.available.createdApps[cx.appID] = struct{}{}
+		if cx.EvalParams.available.createdApps == nil {
+			cx.EvalParams.available.createdApps = make(map[basics.AppIndex]struct{})
 		}
+		cx.EvalParams.available.createdApps[cx.appID] = struct{}{}
```

Comment:
This change is the heart of the matter, instead of only adding the new appID early after the protocol has upgraded to allow resourceSharing, we always add it.